### PR TITLE
Add in code for #VA, #VD, and #VE codes used at XPCS beamlines at APS.

### PR DESCRIPTION
**Pull request by JPHammonds**
I have created a new plugin for codes used at XPCS beamlines at APS.  I have included a test routine and a test spec file used by test and for example.  Would like to add this to the base since this does not seem to overlap any existing spec codes.

### CODE REVIEW

**Comment by prjemian** [comment]
It may be useful to add code that will write out these data in a NeXus file.

**Comment by JPHammonds** [comment]
I have added new code for the writer to add these elements to a NeXus files.

File: src/spec2nexus/plugins/XPCS_spec2nexus.py [hunk 1]
```diff
@@ -1,39 +1,71 @@
...
-        spec_obj.VA.append( strip_first_word(text) )
+            spec_obj.VA = {}
+        spec_obj.VA[subkey] = strip_first_word(text)
+        if isinstance(spec_obj, SpecDataFileHeader):
+            spec_obj.addH5writer(self.key, self.writer)
     
+    def writer(self, h5parent, writer, scan, nxclass=None, *args, **kws):
+        ''' Describe how to write VA data'''
+        desc = "XPCS VA parameters"
+        group = makeGroup(h5parent, 'VA', nxclass,descriuption=desc)
```
Comment (prjemian) [inline src/spec2nexus/plugins/XPCS_spec2nexus.py]: typo: description

File: src/spec2nexus/plugins/XPCS_spec2nexus.py [hunk 2]
```diff
@@ -1,39 +1,71 @@
...
-        spec_obj.VD.append( strip_first_word(text) )
-        #spec_obj.addPostProcessor('unicat_metadata', self.postprocess)
+        spec_obj.VD[subkey] = strip_first_word(text)
+        if isinstance(spec_obj, SpecDataFileHeader):
+            spec_obj.addH5writer(self.key, self.writer)
     
+    def writer(self, h5parent, writer, scan, nxclass=None, *args, **kws):
+        ''' Describe how to write VD data'''
+        desc = "XPCS VD parameters"
+        group = makeGroup(h5parent, 'VD', nxclass, descriuption=desc)
```
Comment (prjemian) [inline src/spec2nexus/plugins/XPCS_spec2nexus.py]: typo: description

File: src/spec2nexus/plugins/XPCS_spec2nexus.py [hunk 3]
```diff
@@ -1,39 +1,71 @@
...
-        spec_obj.VE.append( strip_first_word(text) )
-        #spec_obj.addPostProcessor('unicat_metadata', self.postprocess)
+        spec_obj.VE[subkey] = strip_first_word(text)
+        if isinstance(spec_obj, SpecDataFileHeader):
+            spec_obj.addH5writer(self.key, self.writer)
     
+    def writer(self, h5parent, writer, scan, nxclass=None, *args, **kws):
+        ''' Describe how to write VE data'''
+        desc = "XPCS VE parameters"
+        group = makeGroup(h5parent, 'VE', nxclass,descriuption=desc)
```
Comment (prjemian) [inline src/spec2nexus/plugins/XPCS_spec2nexus.py]: typo: description

**Comment by JPHammonds** [comment]
I have pushed fixes for the typos that were discovered.

**Pull request merged**